pkg/proxy/handlers/spacelister.go

```diff
@@ -81,68 +150,98 @@ func (s *SpaceLister) ListUserWorkspaces(ctx echo.Context) ([]toolchainv1alpha1.
 	}
+	// specific workspace requested so add label requirement to match the space
+	spaceSelector, err := labels.NewRequirement(toolchainv1alpha1.SpaceBindingSpaceLabelKey, selection.Equals, []string{workspaceName})
+	if err != nil {
+		return nil, err
+	}
+	requirements := []labels.Requirement{*murSelector, *spaceSelector}
 
-	requirements := []labels.Requirement{*murSelector}
+	spaceBindings, err := s.GetInformerServiceFunc().ListSpaceBindings(requirements...)
+	if err != nil {
+		return nil, err
+	}
 
-	if doGetWorkspace {
-		// specific workspace requested so add label requirement to match the space
-		spaceSelector, err := labels.NewRequirement(toolchainv1alpha1.SpaceBindingSpaceLabelKey, selection.Equals, []string{workspaceName})
-		if err != nil {
-			return nil, err
-		}
-		requirements = append(requirements, *spaceSelector)
+	//  let's only log the issue and consider this as not found
+	if len(spaceBindings) != 1 {
+		ctx.Logger().Error("expected only 1 spacebinding, got %d for user %s and workspace %s", len(spaceBindings), murName, workspaceName)
+		return nil, nil
 	}
 
-	return s.GetInformerServiceFunc().ListSpaceBindings(requirements...)
+	return &spaceBindings[0], nil
 }
 
 func (s *SpaceLister) workspacesFromSpaceBindings(signupName string, spaceBindings []toolchainv1alpha1.SpaceBinding) []toolchainv1alpha1.Workspace {
 	workspaces := []toolchainv1alpha1.Workspace{}
-	for _, spaceBinding := range spaceBindings {
-
-		var ownerName string
-		spaceName := spaceBinding.Labels[toolchainv1alpha1.SpaceBindingSpaceLabelKey]
-		if spaceName == "" { // space may not be initialized
-			// log error and continue so that the api behaves in a best effort manner
-			log.Errorf(nil, fmt.Errorf("spacebinding has no '%s' label", toolchainv1alpha1.SpaceBindingSpaceLabelKey), "unable to get space", "space", spaceName)
-			continue
-		}
-		space, err := s.GetInformerServiceFunc().GetSpace(spaceName)
+	for i := range spaceBindings {
+		spacebinding := &spaceBindings[i]
+		space, err := s.getSpace(spacebinding)
 		if err != nil {
 			// log error and continue so that the api behaves in a best effort manner
 			// ie. if a space isn't listed something went wrong but we still want to return the other spaces if possible
-			log.Errorf(nil, err, "unable to get space", "space", spaceName)
+			log.Errorf(nil, err, "unable to get space", "space", spacebinding.Labels[toolchainv1alpha1.SpaceBindingSpaceLabelKey])
 			continue
 		}
+		workspace := createCommonWorkspace(signupName, space, spacebinding)
+		workspaces = append(workspaces, *workspace)
+	}
+	return workspaces
+}
 
-		// TODO right now we get SpaceCreatorLabelKey but should get owner from Space once it's implemented
-		ownerName = space.Labels[toolchainv1alpha1.SpaceCreatorLabelKey]
+func createCommonWorkspace(signupName string, space *toolchainv1alpha1.Space, spaceBinding *toolchainv1alpha1.SpaceBinding, wsAdditionalOptions ...commonproxy.WorkspaceOption) *toolchainv1alpha1.Workspace {
```

Comment:
Maybe something like `createWorkspaceObject()` would be a better name?

Reply:
yeah, I was unsure about a proper name (as always 😅 ). thanks for the suggestion! 

Applied: https://github.com/codeready-toolchain/registration-service/pull/348/commits/e83efa484c4ca06b4053e640407cb188ae1e7fdb